synapse/config/server.py

```diff
@@ -54,6 +54,10 @@ def read_config(self, config):
         self.public_baseurl = config.get("public_baseurl")
         self.cpu_affinity = config.get("cpu_affinity")
 
+        # Whether to use the asynchronous DNS resolver instead of getaddrinfo.
+        # This can potentially improve performance.
+        self.no_getaddrinfo = config.get("no_getaddrinfo", False)
```

Comment:
Before I forget, can we make sure we remove the `no_` prefix please? It can get very confusing when you have things like `no_getaddrinfo: true`, and we've been trying to get rid of them from the code base. Something like `use_getaddrinfo_for_dns` or something maybe?

(Sorry for the bike shedding without reviewing the rest of the PR, but I'll leave that to @hawkowl )

Reply:
For documentation purposes it'd also be good to put this in the default_config below too, probably commented out:

```yaml
# Whether to use the default system resolver (getaddrinfo) or to use the twisted dns resolver.
# Defaults to True. Uncomment to use twisted resolver instead.
#use_getaddrinfo_for_dns: False.
```

but with better words